pandas/core/categorical.py

```diff
@@ -1851,7 +1851,7 @@ def unique(self):
         """
 
         # unlike np.unique, unique1d does not sort
-        unique_codes = unique1d(self.codes)
+        unique_codes = unique1d(self.codes).astype(self.codes.dtype)
```

Comment:
In case of groupby it will be upcast, but in general, .unique() should be type preserving?